lib/readline.js

```diff
@@ -873,6 +859,84 @@ Interface.prototype._ttyWrite = function(s, key) {
 
 exports.Interface = Interface;
 
+function ReadlineHistory(options) {
+  this._maxEntries = options.maxEntries;
+  this._index = 0;
+  this._entries = [];
+}
+
+ReadlineHistory.prototype.addEntry = function(line) {
+  var added = false;
+  if (line && this._entries[this._entries.length - 1] !== line) {
+    if (this._maxEntries == this._entries.length) {
+      this._entries.shift();
+    }
+    this._entries.push(line);
+    added = true;
+  }
+  this._index = this._entries.length;
+  return added;
+};
+
+ReadlineHistory.prototype.nextEntry = function() {
+  this._index += 1;
+  if (this._index >= this._entries.length) {
+    this._index = this._entries.length;
+    return '';
+  }
+  return this._entries[this._index];
+};
+
+ReadlineHistory.prototype.prevEntry = function() {
+  this._index -= 1;
+  if (this._index < 0) this._index = 0;
+  return this._entries[this._index];
+};
+
+function ReadlineFileHistory(options) {
+  if (!path) path = require('path');
+  if (!fs) fs = require('fs');
```

Comment:
I don't see any benefit to lazy loading these modules. Could you hoist this declaration to the top?